<?xml version="1.0"?>
<robot name="/home/ayrtom/Computer Vision/dev/FQW_ws/src/surround-view-segbev/resources/descriptions/EgoVehicle.urdf" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <origin xyz="1.69 0 0.52" rpy="0 0 0"/>
      <geometry>
        <box size="6.6 2 1"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="1.69 0 0.52" rpy="0 0 0"/>
      <geometry>
        <box size="6.6 2 1"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0.67 0 1.64" rpy="0 0 0"/>
      <geometry>
        <box size="4.5 1.95 1.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0.67 0 1.64" rpy="0 0 0"/>
      <geometry>
        <box size="4.5 1.95 1.3"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="3.60031 0 1.09128" rpy="0 0.55 0"/>
      <geometry>
        <box size="2.5 1.88 1.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="3.60031 0 1.09128" rpy="0 0.55 0"/>
      <geometry>
        <box size="2.5 1.88 1.3"/>
      </geometry>
    </collision>
  </link>
  <link name="backwards_lights">
  </link>
  <joint name="base_link_backwards_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="backwards_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="rear_lights">
  </link>
  <joint name="base_link_rear_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="rear_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="brake_lights">
  </link>
  <joint name="base_link_brake_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="brake_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="antifog_lights">
  </link>
  <joint name="base_link_antifog_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="antifog_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="left_indicators">
  </link>
  <joint name="base_link_left_indicators_joint" type="fixed">
    <parent link="base_link"/>
    <child link="left_indicators"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="right_indicators">
  </link>
  <joint name="base_link_right_indicators_joint" type="fixed">
    <parent link="base_link"/>
    <child link="right_indicators"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="front_lights">
  </link>
  <joint name="base_link_front_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="front_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="camera_rear">
  </link>
  <joint name="base_link_camera_rear_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_rear"/>
    <origin xyz="-1.613499 0.007999 0.555" rpy="-1.570796 0 1.570704"/>
  </joint>
  <link name="camera_front">
  </link>
  <joint name="base_link_camera_front_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_front"/>
    <origin xyz="5.26 0.01 0.48" rpy="-1.570796 0 -1.570796"/>
  </joint>
  <link name="engine_speaker">
  </link>
  <joint name="base_link_engine_speaker_joint" type="fixed">
    <parent link="base_link"/>
    <child link="engine_speaker"/>
    <origin xyz="4.325 0 0" rpy="0 0 0"/>
  </joint>
  <joint name="right_rear_sensor" type="continuous">
    <parent link="base_link"/>
    <child link="rear right wheel"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0 -0.85 0" rpy="0 -0.000097 0"/>
  </joint>
  <link name="rear right wheel">
    <visual>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <geometry>
        <cylinder radius="0.4" length="0.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <geometry>
        <cylinder radius="0.4" length="0.3"/>
      </geometry>
    </collision>
  </link>
  <joint name="slot_rear right wheel_joint" type="fixed">
    <parent link="base_link"/>
    <child link="rear right wheel"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <joint name="left_rear_sensor" type="continuous">
    <parent link="base_link"/>
    <child link="rear left wheel"/>
    <axis xyz="0 -1 0"/>
    <origin xyz="0 0.85 0" rpy="0 0.000097 3.14159"/>
  </joint>
  <link name="rear left wheel">
    <visual>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <geometry>
        <cylinder radius="0.4" length="0.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <geometry>
        <cylinder radius="0.4" length="0.3"/>
      </geometry>
    </collision>
  </link>
  <joint name="slot_0_rear left wheel_joint" type="fixed">
    <parent link="base_link"/>
    <child link="rear left wheel"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
</robot>
